<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#futures" data-toc-modified-id="futures-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>futures</a></span></li><li><span><a href="#und-dann-alle-usdt-auf-LONG-BTC-gesetzt-werden" data-toc-modified-id="und-dann-alle-usdt-auf-LONG-BTC-gesetzt-werden-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>und dann alle usdt auf LONG BTC gesetzt werden</a></span></li><li><span><a href="#und-dann-alle-usdt-auf-LONG-BTC-gesetzt-werden" data-toc-modified-id="und-dann-alle-usdt-auf-LONG-BTC-gesetzt-werden-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>und dann alle usdt auf LONG BTC gesetzt werden</a></span></li></ul></div>

In [2]:
api_key = "" 
api_secret = ""

In [3]:
from binance.client import Client
import time
client = Client(api_key, api_secret)

usdt_acc_balance:  3271.14431913


In [5]:
client.futures_position_information()

[{'symbol': 'EOSUSDT',
  'positionAmt': '0.0',
  'entryPrice': '0.0000',
  'markPrice': '0.00000000',
  'unRealizedProfit': '0.00000000',
  'liquidationPrice': '0',
  'leverage': '20',
  'maxNotionalValue': '250000',
  'marginType': 'cross',
  'isolatedMargin': '0.00000000',
  'isAutoAddMargin': 'false',
  'positionSide': 'BOTH'},
 {'symbol': 'EOSUSDT',
  'positionAmt': '0.0',
  'entryPrice': '0.0000',
  'markPrice': '0.00000000',
  'unRealizedProfit': '0.00000000',
  'liquidationPrice': '0',
  'leverage': '20',
  'maxNotionalValue': '250000',
  'marginType': 'cross',
  'isolatedMargin': '0.00000000',
  'isAutoAddMargin': 'false',
  'positionSide': 'LONG'},
 {'symbol': 'EOSUSDT',
  'positionAmt': '0.0',
  'entryPrice': '0.0000',
  'markPrice': '0.00000000',
  'unRealizedProfit': '0.00000000',
  'liquidationPrice': '0',
  'leverage': '20',
  'maxNotionalValue': '250000',
  'marginType': 'cross',
  'isolatedMargin': '0.00000000',
  'isAutoAddMargin': 'false',
  'positionSide': 'SHORT'},


In [6]:
# get market depth
depth = client.get_order_book(symbol='BNBBTC')

# place a test market buy order, to place an actual order use the create_order function
order = client.create_test_order(
    symbol='BNBBTC',
    side=Client.SIDE_BUY,
    type=Client.ORDER_TYPE_MARKET,
    quantity=100)
order

{}

In [7]:
# get all symbol prices
prices = client.get_all_tickers()
prices

[{'symbol': 'ETHBTC', 'price': '0.03294600'},
 {'symbol': 'LTCBTC', 'price': '0.00476900'},
 {'symbol': 'BNBBTC', 'price': '0.00165820'},
 {'symbol': 'NEOBTC', 'price': '0.00102000'},
 {'symbol': 'QTUMETH', 'price': '0.00455900'},
 {'symbol': 'EOSETH', 'price': '0.00553500'},
 {'symbol': 'SNTETH', 'price': '0.00006199'},
 {'symbol': 'BNTETH', 'price': '0.00184600'},
 {'symbol': 'BCCBTC', 'price': '0.07908100'},
 {'symbol': 'GASBTC', 'price': '0.00007790'},
 {'symbol': 'BNBETH', 'price': '0.05030200'},
 {'symbol': 'BTCUSDT', 'price': '18326.77000000'},
 {'symbol': 'ETHUSDT', 'price': '604.00000000'},
 {'symbol': 'HSRBTC', 'price': '0.00041400'},
 {'symbol': 'OAXETH', 'price': '0.00017780'},
 {'symbol': 'DNTETH', 'price': '0.00002801'},
 {'symbol': 'MCOETH', 'price': '0.00577200'},
 {'symbol': 'ICNETH', 'price': '0.00166300'},
 {'symbol': 'MCOBTC', 'price': '0.00021140'},
 {'symbol': 'WTCBTC', 'price': '0.00002110'},
 {'symbol': 'WTCETH', 'price': '0.00063900'},
 {'symbol': 'LRCBTC', 'pr

In [8]:
avg_price = client.get_avg_price(symbol='BTCUSDT')
avg_price

{'mins': 5, 'price': '18308.21366192'}

In [9]:
#Get 24hr Ticker
client.get_ticker(symbol='BTCUSDT')

{'symbol': 'BTCUSDT',
 'priceChange': '-131.67000000',
 'priceChangePercent': '-0.713',
 'weightedAvgPrice': '18433.87952696',
 'prevClosePrice': '18458.80000000',
 'lastPrice': '18327.14000000',
 'lastQty': '0.00277200',
 'bidPrice': '18327.14000000',
 'bidQty': '0.03063100',
 'askPrice': '18327.15000000',
 'askQty': '0.19016800',
 'openPrice': '18458.81000000',
 'highPrice': '18766.00000000',
 'lowPrice': '18000.00000000',
 'volume': '83908.76514000',
 'quoteVolume': '1546764067.84638922',
 'openTime': 1606086848782,
 'closeTime': 1606173248782,
 'firstId': 477359619,
 'lastId': 478988973,
 'count': 1629355}

# futures

In [20]:
current_btc_price_info = client.futures_mark_price(symbol='BTCUSDT')
current_btc_price = float(current_btc_price_info["markPrice"])
current_btc_price

18308.74407753

In [11]:
active_futures_list = client.futures_position_information(symbol='BTCUSDT')
active_futures_list

[{'symbol': 'BTCUSDT',
  'positionAmt': '0.000',
  'entryPrice': '0.00000',
  'markPrice': '18339.78318596',
  'unRealizedProfit': '0.00000000',
  'liquidationPrice': '0',
  'leverage': '1',
  'maxNotionalValue': '9223372036854775807',
  'marginType': 'isolated',
  'isolatedMargin': '0.00000000',
  'isAutoAddMargin': 'false',
  'positionSide': 'BOTH'},
 {'symbol': 'BTCUSDT',
  'positionAmt': '0.176',
  'entryPrice': '18547.30000',
  'markPrice': '18339.78318596',
  'unRealizedProfit': '-36.52295927',
  'liquidationPrice': '48.57112820',
  'leverage': '1',
  'maxNotionalValue': '9223372036854775807',
  'marginType': 'isolated',
  'isolatedMargin': '3219.28751624',
  'isAutoAddMargin': 'false',
  'positionSide': 'LONG'},
 {'symbol': 'BTCUSDT',
  'positionAmt': '0.000',
  'entryPrice': '0.00000',
  'markPrice': '18339.78318596',
  'unRealizedProfit': '0.00000000',
  'liquidationPrice': '0',
  'leverage': '1',
  'maxNotionalValue': '9223372036854775807',
  'marginType': 'isolated',
  'isol

In [12]:
# Hier der Preis der oben bei binance angeeigt wird

active_futures_list = client.futures_position_information(symbol='BTCUSDT')

for x in range( len(active_futures_list)):
    if active_futures_list[x]['symbol'] == 'BTCUSDT' and float(active_futures_list[x]["positionAmt"]) >= 0.000001:
        current_short_or_long = active_futures_list[x]["positionSide"]
        print(current_short_or_long)
        
current_short_or_long

LONG


'LONG'

In [13]:
current_short_or_long

'LONG'

In [14]:
# Hier der Preis der oben bei binance angeeigt wird
entryprice_btc_list = []

active_futures_list = client.futures_position_information(symbol='BTCUSDT')

for x in range( len(active_futures_list)):
    if active_futures_list[x]['symbol'] == 'BTCUSDT' and float(active_futures_list[x]["positionAmt"]) >= 0.000001:
        print(float(active_futures_list[x]["entryPrice"]))
        entryprice_btc_list.append(float(active_futures_list[x]["entryPrice"]))

sum(entryprice_btc_list)

18547.3


18547.3

In [94]:
# Hier der Preis der oben bei binance angeeigt wird
money_longshort_btc_list = []

active_futures_list = client.futures_position_information(symbol='BTCUSDT')

for x in range( len(active_futures_list)):
    if active_futures_list[x]['symbol'] == 'BTCUSDT' and float(active_futures_list[x]["positionAmt"]) >= 0.000001:
        print(float(active_futures_list[x]["positionAmt"]))
        money_longshort_btc_list.append(float(active_futures_list[x]["positionAmt"]))
 
money_longshort_btc_list

0.176


[0.176]

In [95]:
sum_money_longshort_btc = sum(money_longshort_btc_list)
sum_money_longshort_btc

0.176

In [49]:
#client.futures_get_order(symbol = 'BTCUSDT')

In [62]:
# all_orders = client.futures_get_all_orders()
# all_orders

In [63]:
number_of_all_orders = len(all_orders)
number_of_all_orders

3

In [65]:
# to cancel multiple orders symbol is necesary!
all_orders[0]["symbol"]

'BTCUSDT'

In [68]:
# all_orders[0]

In [55]:
# IMPORTANT: Set leverage to 1x!
client.futures_change_leverage(symbol='BTCUSDT', leverage=1)
client.futures_change_leverage(symbol='ETHUSDT', leverage=1)

{'symbol': 'BTCUSDT', 'leverage': 1, 'maxNotionalValue': 'INF'}

In [58]:
def future_order(symbol, positionSide, quantity, side):
    try:
        order = client.futures_create_order(symbol=symbol,
                                            positionSide=positionSide,
                                            quantity=quantity,
                                            side=side,
                                            type=Client.ORDER_TYPE_MARKET,
                                            )
    except Exception as e:
        print("an exception occurred - {}".format(e))
        return False
    return True 

# und dann alle usdt auf LONG BTC gesetzt werden
quantityBTC=0.176
future_order('BTCUSDT', "LONG", quantityBTC, Client.SIDE_BUY)

# und dann alle usdt auf LONG BTC gesetzt werden
quantityBTC=0.076
future_order('BTCUSDT', "LONG", quantityBTC, Client.SIDE_SELL)

In [10]:
active_futures_list = client.futures_position_information(symbol='BTCUSDT')

for x in range( len(active_futures_list)):
    if active_futures_list[x]['symbol'] == 'BTCUSDT' and float(active_futures_list[x]["positionAmt"]) >= 0.000001:
        current_short_or_long = active_futures_list[x]["positionSide"]
        print(current_short_or_long)


LONG


'LONG'

In [21]:
# get money in tha bank
# get my current account balance
futures_acc_balance_list = client.futures_account_balance()
for x in range( len(futures_acc_balance_list)) :
    if futures_acc_balance_list[x]["asset"] == 'USDT':
        usdt_acc_balance = futures_acc_balance_list[x]["balance"]
        usdt_acc_balance = float(usdt_acc_balance)
        print("usdt_acc_balance: ", usdt_acc_balance)

# current btc futures price
current_btc_price_info = client.futures_mark_price(symbol='BTCUSDT')
current_btc_price = float(current_btc_price_info["markPrice"])

# get affordable btc to buy 
btcs_to_set = round(usdt_acc_balance/current_btc_price, 3)
btcs_to_set


usdt_acc_balance:  3271.14431913


3271.14431913

0.178

0.178

In [ ]:
# Wenn hier 0 als input kommt vom ml modell dann müssen  
if algoprediction = 0:
    # check ob wir LONG BTC orders haben??
    if current_short_or_long == 'LONG':    
        # alle LONG BTC orders verkauft werden (Check ob alle verkauft)
        future_order('BTCUSDT', "LONG", sum_money_longshort_btc, Client.SIDE_SELL)
        time.sleep(5)
        
        # get money in tha bank
        # get my current account balance
        futures_acc_balance_list = client.futures_account_balance()
        for x in range( len(futures_acc_balance_list)) :
            if futures_acc_balance_list[x]["asset"] == 'USDT':
                usdt_acc_balance = futures_acc_balance_list[x]["balance"]
                usdt_acc_balance = float(usdt_acc_balance)
                print("usdt_acc_balance: ", usdt_acc_balance)

        # current btc futures price
        current_btc_price_info = client.futures_mark_price(symbol='BTCUSDT')
        current_btc_price = float(current_btc_price_info["markPrice"])

        # get affordable btc to buy 
        btcs_to_set = round(usdt_acc_balance/current_btc_price, 3)
        
        time.sleep(1)
        
        # und dann alle usdt auf SHORT BTC gesetzt werden
        future_order('BTCUSDT', "SHORT", btcs_to_set, Client.SIDE_BUY)
        
# Wenn hier 1 als input kommt vom ml modell dann müssen  
elif algoprediction = 1:
    # check ob wir SHORT BTC orders haben??
    if current_short_or_long == 'SHORT':        
        # alle SHORT BTC orders verkauft werden (Check ob alle verkauft)
        future_order('BTCUSDT', "SHORT", sum_money_longshort_btc, Client.SIDE_SELL)
        time.sleep(5)
        
        # get money in tha bank
        # get my current account balance
        futures_acc_balance_list = client.futures_account_balance()
        for x in range( len(futures_acc_balance_list)) :
            if futures_acc_balance_list[x]["asset"] == 'USDT':
                usdt_acc_balance = futures_acc_balance_list[x]["balance"]
                usdt_acc_balance = float(usdt_acc_balance)
                print("usdt_acc_balance: ", usdt_acc_balance)

        # current btc futures price
        current_btc_price_info = client.futures_mark_price(symbol='BTCUSDT')
        current_btc_price = float(current_btc_price_info["markPrice"])

        # get affordable btc to buy 
        btcs_to_set = round(usdt_acc_balance/current_btc_price, 3)
        
        time.sleep(1)
        
        # und dann alle usdt auf LONG BTC gesetzt werden
        future_order('BTCUSDT', "LONG", btcs_to_set, Client.SIDE_BUY)


In [108]:
d = datetime.datetime(2020, 5, 17)
dateofprediction = [d]

In [109]:
if dateofprediction[0] + datetime.timedelta(hours=27) < datetime.datetime.now():
    print("ok")

TypeError: can't compare datetime.datetime to datetime.date